docs/standard/parallel-programming/how-to-create-pre-computed-tasks.md

```diff
@@ -1,27 +1,28 @@
-  
- [!code-csharp[TPL_CachedDownloads#1](../../../samples/snippets/csharp/VS_Snippets_Misc/tpl_cacheddownloads/cs/cacheddownloads.cs#1)]
- [!code-vb[TPL_CachedDownloads#1](../../../samples/snippets/visualbasic/VS_Snippets_Misc/tpl_cacheddownloads/vb/cacheddownloads.vb#1)]  
-  
- This example computes the time that is required to download multiple strings two times. The second set of download operations should take less time than the first set because the results are held in the cache. The <xref:System.Threading.Tasks.Task.FromResult%2A> method enables the `DownloadStringAsync` method to create <xref:System.Threading.Tasks.Task%601> objects that hold these pre-computed results.  
-  
+
+# Create pre-computed tasks
+
+In this article, you'll learn how to use the <xref:System.Threading.Tasks.Task.FromResult%2A?displayProperty=nameWithType> method to retrieve the results of asynchronous download operations that are held in a cache. The <xref:System.Threading.Tasks.Task.FromResult%2A> method returns a finished <xref:System.Threading.Tasks.Task%601> object that holds the provided value as its <xref:System.Threading.Tasks.Task%601.Result%2A> property. This method is useful when you perform an asynchronous operation that returns a <xref:System.Threading.Tasks.Task%601> object, and the result of that <xref:System.Threading.Tasks.Task%601> object is already computed.
+
+## Example
+
+The following example downloads strings from the web. It defines the `DownloadStringAsync` method. This method downloads strings from the web asynchronously. This example also uses a <xref:System.Collections.Concurrent.ConcurrentDictionary%602> object to cache the results of previous operations. If the input address is held in this cache, `DownloadStringAsync` uses the <xref:System.Threading.Tasks.Task.FromResult%2A> method to produce a <xref:System.Threading.Tasks.Task%601> object that holds the content at that address. Otherwise, `DownloadStringAsync` downloads the file from the web and adds the result to the cache.
+
+:::code language="csharp" source="snippets/cs/DownloadCache.cs":::
+:::code language="vb" source="snippets/vb/DownloadCache.vb":::
+
+In the preceding example, the first time each url is downloaded and its value is stored in the cache. The <xref:System.Threading.Tasks.Task.FromResult%2A> method enables the `DownloadStringAsync` method to create <xref:System.Threading.Tasks.Task%601> objects that hold these pre-computed results. Subsequent calls to download the string return the cached values, and is much faster.
```

Comment:
> In the preceding example, the first time each url is downloaded and its value is stored in the cache.

Not a native English speaker **so I might be wrong**, but something grammatically feels wrong in the above statement. Please ignore this comment if you believe its grammatically correct on its current form.

```suggestion
In the preceding example, the first time each url is downloaded, its value is stored in the cache. The <xref:System.Threading.Tasks.Task.FromResult%2A> method enables the `DownloadStringAsync` method to create <xref:System.Threading.Tasks.Task%601> objects that hold these pre-computed results. Subsequent calls to download the string return the cached values, and is much faster.
```

Reply:
I agree with @Youssef1313 That wording is better.